File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.cpp
class Solution {
public:
    int subarraySum(vector<int>& nums, long long k) {
        int res = 0 ;
        for(int i =  0 ; i < nums.size() ; i++){
            int sum = nums[i]; 
            if(sum  == k){
                res++;
            }

            for(int j=i+1; j < nums.size() ; j++){
                sum += nums[j];
                if(sum == k){
                    res++;
                }  
                
               
            
            }
        }
        return res;
    }
};

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int ka) {
        int res = 0 ;
        int prev = 0 ;
        HashMap<Integer , Integer> seen = new HashMap<>() ;
        seen.put(0,1) ;

        

        for(int i = 0 ; i < nums.length ; i++){
            prev += nums[i] ;
            int k = prev - ka ;

            if(seen.containsKey(k)){
                res += seen.get(k) ;
            }

            seen.put(prev , seen.getOrDefault(prev,0) + 1);

        }

        return res ;
    }
}